WooCommerce/WooCommerceTests/Extensions/DateWooTests.swift

```diff
@@ -4,81 +4,108 @@ import XCTest
 
 /// Date+Woo: Unit Tests
 ///
-class DateWooTests: XCTestCase {
+final class DateWooTests: XCTestCase {
 
     func testUpdateStringWorksForIntervalsUnderTwoMinutes() {
 
         // 1 second
+        let momentsAgo = NSLocalizedString("Updated moments ago",
+                                           comment: "A unit test string for relative time intervals")
         let oneSecondAgo = Calendar.current.date(byAdding: .second, value: -1, to: Date())!
-        XCTAssertEqual(oneSecondAgo.relativelyFormattedUpdateString, "Updated moments ago")
+        XCTAssertEqual(oneSecondAgo.relativelyFormattedUpdateString, momentsAgo)
 
         // 12 seconds
         let twelveSecondsAgo = Calendar.current.date(byAdding: .second, value: -12, to: Date())!
-        XCTAssertEqual(twelveSecondsAgo.relativelyFormattedUpdateString, "Updated moments ago")
+        XCTAssertEqual(twelveSecondsAgo.relativelyFormattedUpdateString, momentsAgo)
 
         // 1 minute, 59 seconds
         let almostTwoMinutesAgo = Calendar.current.date(byAdding: .second, value: -119, to: Date())!
-        XCTAssertEqual(almostTwoMinutesAgo.relativelyFormattedUpdateString, "Updated moments ago")
+        XCTAssertEqual(almostTwoMinutesAgo.relativelyFormattedUpdateString, momentsAgo)
 
         // 2 minutes
         let twoMinutesAgo = Calendar.current.date(byAdding: .minute, value: -2, to: Date())!
-        XCTAssertNotEqual(twoMinutesAgo.relativelyFormattedUpdateString, "Updated moments ago")
+        XCTAssertNotEqual(twoMinutesAgo.relativelyFormattedUpdateString, momentsAgo)
     }
 
     func testUpdateStringWorksForIntervalsOneDayOrLess() {
 
+        let minutesAgo = NSLocalizedString("Updated 2 minutes ago",
+                                           comment: "A unit test string for time intervals")
+        let almostHourAgo = NSLocalizedString("Updated 59 minutes ago",
+                                              comment: "A unit test string for a plural time interval in minutes")
+        let hourAgo = NSLocalizedString("Updated 1 hour ago",
+                                        comment: "A unit test string for a singular time interval")
+        let nineAgo = NSLocalizedString("Updated 9 hours ago",
+                                        comment: "A unit test string for a plural time interval in hours")
+        let almostDayAgo = NSLocalizedString("Updated 23 hours ago",
+                                             comment: "A unit test string for time interval just under 1 day")
+        let dayAgo = NSLocalizedString("Updated 24 hours ago",
```

Comment:
I'd suggest using "Updated %ld hours ago" format (and its variants in `Localizable.strings`) and `String.localizedStringWithFormat` to create the expected string. these strings "Updated # minutes/hours ago" have the number hardcoded, and because they don't exist in `Localizable.strings` these strings aren't localizable and will always be in English.

Reply:
The current expectation is that these 8 unit tests in DateWooTests will fail. They will pass once the NSLocalizedStrings have been translated. They will exist in `Localizable.strings` once this PR is merged and a build is created. Part of the build process is to run the script and find new strings that need localized and set them up for translation.

The point of creating the localized strings with the times hardcoded is to have a fully translated string that was not formatted or modified by the `relativelyFormattedString` method. That way, a "hardcoded string" for the expected result is compared to the actual result from the `relativelyFormattedString` method. If I create a `NSLocalizedString` that added an argument for the minutes and hours, it would be incorrect because it doesn't account for pluralizations. If I further modified the `NSLocalizedString` to account for the pluralization, then all I am doing is re-creating the `relativelyFormattedUpdateString` method, when what I really want is to compare the actual output to the expected output.

Does that make sense?

Reply:
I feel it's a bit odd to have new localizable strings from the unit tests in general (there might be exceptions), I'm not sure if we want to put the (community) effort localizing strings just for internal unit tests. This also creates a dependency for our unit tests on the external translation - what if the hard-coded strings are translated to a different string from the localized format + value? If we use the existing localizable strings in unit tests, even if some strings haven't been translated yet they will at least match and pass in tests.

Looking at this test class, it seems to me that it's trying to test the "Updated ... ago" string with different units and unit plural formats given different timestamps. I feel like testing with the localized format + value would achieve the goals of this test case.

@pmusolino what do you think?

Reply:
I think these unit tests have the same problem as [this one](https://github.com/woocommerce/woocommerce-ios/pull/1552/commits/af6fc028c0389a07b489fd2452392e252b3f152e). Would it be better to remove these altogether? 

> (community) effort

As an aside, our translations are done by a professional service. I would consider it money well spent if the unit tests are alerting us to a format that has been altered and doesn't suit multiple languages. (That is, if these unit tests are useful in the first place.)

Reply:
In doing so, is there not a risk that the work of the translators will affect the test result?